executor/runtime/docker/capabilities.go

```diff
@@ -0,0 +1,68 @@
+package docker
+
+import (
+	"github.com/Netflix/titus-executor/executor/runtime/docker/seccomp"
+	runtimeTypes "github.com/Netflix/titus-executor/executor/runtime/types"
+	"github.com/docker/docker/api/types/container"
+
+	"fmt"
+)
+
+func addAdditionalCapabilities(c *runtimeTypes.Container, hostCfg *container.HostConfig) map[string]struct{} {
+	addedCapabilities := make(map[string]struct{})
+
+	// Set any additional capabilities for this container
+	if cap := c.TitusInfo.GetCapabilities(); cap != nil {
+		for _, add := range cap.GetAdd() {
+			addedCapabilities[add.String()] = struct{}{}
+			hostCfg.CapAdd = append(hostCfg.CapAdd, add.String())
+		}
+		for _, drop := range cap.GetDrop() {
+			hostCfg.CapDrop = append(hostCfg.CapDrop, drop.String())
+		}
+	}
+	return addedCapabilities
+}
+
+func setupAdditionalCapabilities(c *runtimeTypes.Container, hostCfg *container.HostConfig) error {
+	addedCapabilities := addAdditionalCapabilities(c, hostCfg)
+
+	// Privileged containers automaticaly deactivate seccomp and friends, no need to do this
+	fuseEnabled, err := c.GetFuseEnabled()
+	if err != nil {
+		return err
+	}
+	if fuseEnabled || c.TitusInfo.GetAllowNestedContainers() {
+		if _, ok := addedCapabilities["SYS_ADMIN"]; !ok {
+			hostCfg.CapAdd = append(hostCfg.CapAdd, "SYS_ADMIN")
+		}
+	}
+	seccompProfile := "default.json"
+	apparmorProfile := ""
+
+	if fuseEnabled {
+		hostCfg.Resources.Devices = append(hostCfg.Resources.Devices, container.DeviceMapping{
+			PathOnHost:        fuseDev,
+			PathInContainer:   fuseDev,
+			CgroupPermissions: "rmw",
+		})
+		apparmorProfile = "docker-fuse"
+		seccompProfile = "fuse-container.json"
+
+	}
+	// We can do this here because nested containers can do everything fuse containers can
+	if c.TitusInfo.GetAllowNestedContainers() {
+		apparmorProfile = "docker-nested"
```

Comment:
do the nested apparmor and seccomp profiles include everything that's in the fuse ones? (in case both `FUSE && nestedContainers` is enabled for a task)

Reply:
@fabiokung Yeah, I thought the comment indicated that: `	// We can do this here because nested containers can do everything fuse containers can` -- The capabilities of nested containers are a superset of fuse containers.

Reply:
ugh, I missed that comment. LGTM